src/PaginationCommand.php

```diff
@@ -7,63 +7,87 @@
+use PhpCollection\Sequence;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 
-abstract class PaginationCommand extends Command {
+abstract
+class PaginationCommand extends Command {
     use WithClient;
 
     protected
     function configure() {
         $this
-            ->addOption("debug","d", InputOption::VALUE_REQUIRED, false)
+            ->addOption("debug", "d", InputOption::VALUE_REQUIRED, false)
             ->addOption("pages", "p", InputOption::VALUE_OPTIONAL, 'number of pages to retrieve', 10)
-            ->addOption("size", "s", InputOption::VALUE_OPTIONAL, 'page size', 3);
+            ->addOption("size", "s", InputOption::VALUE_OPTIONAL, 'page size', 3)
+            ->addOption("filter", "f", InputOption::VALUE_OPTIONAL, 'page size', '');
     }
 
     protected
-    abstract function get_endpoint();
+    abstract
+    function get_endpoint();
 
-    protected
-    function execute(InputInterface $input, OutputInterface $output) {
+    public
+    function getResult(InputInterface $input, OutputInterface $output) {
         try {
-            $page_num = 1;
-            $client = $this->getClient();
-            $page_size = $input->getOption('size');
+            $items = new Sequence();
+
+            $page_num     = 1;
+            $client       = $this->getClient();
+            $page_size    = $input->getOption('size');
             $page_num_max = $input->getOption('pages');
+            $filter       = $input->getOption('filter');
 
             if ($page_size > 25) {
                 $output->writeln("$page_size exceeds page size limit 25, setting it to 25 ...");
                 $page_size = 25;
             }
 
             $url = $this->get_endpoint();
-            $r = $client->get($url, ['query' => ['limit' => $page_size]]);
+            $r   = $client->get($url, [
+                'query'               => [
+                    'limit'  => $page_size,
+                    'filter' => $filter
+                ],
+                RequestOptions::DEBUG =>
+                    boolval($input->getOption('debug')) ? fopen('php://stdout',
+                                                                'w') : false
+            ]);
+
             $result = json_decode($r->getBody()->getContents());
-            $this->printResult($page_num, $result, $output);
+            $items->addAll($result->items);
+
 
             while (isset($result->links) && !empty($result->links->next) && $page_num < $page_num_max) {
                 $url = $result->links->next;
-                $r = $client->get($url);
+                $r   = $client->get($url);
                 $page_num++;
                 $result = json_decode($r->getBody()->getContents());
-                $this->printResult($page_num, $result, $output);
+                $items->addAll($result->items);
```

Comment:
This would probably cause out of memory if $page_num_max is set to a high number